hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/wal/SyncFuture.java

```diff
@@ -44,19 +45,22 @@
  */
 @InterfaceAudience.Private
 class SyncFuture {
-  // Implementation notes: I tried using a cyclicbarrier in here for handler and sync threads
-  // to coordinate on but it did not give any obvious advantage and some issues with order in which
-  // events happen.
+
   private static final long NOT_DONE = -1L;
+  private Thread t;
 
-  /**
-   * The transaction id of this operation, monotonically increases.
-   */
-  private long txid;
+  // Lock protecting the thread safe fields
+  ReentrantLock doneLock;
+  // Condition to wait on for client threads.
+  Condition isDone;
```

Comment:
nit: isDone is name of a method, not of a data member.

Reply:
And name it something else because isDone method already?

Reply:
doneCondition?

Reply:
Ack, done.